libvast/src/experimental_table_slice_builder.cpp

```diff
@@ -689,21 +691,81 @@ bool experimental_table_slice_builder::add_impl(data_view x) {
 
 // -- utility functions --------------------------------------------------------
 
+namespace {
+
+std::vector<struct type::attribute>
+deserialize_attributes(std::string_view serialized) {
+  simdjson::padded_string json{serialized};
+  simdjson::dom::parser parser;
+  auto doc = parser.parse(json);
+  // TODO: use field_view as soon as simdjson::ondemand hits (one less copy)
+  std::vector<struct type::attribute> attrs{};
+  for (auto f : doc.get_object()) {
+    auto value = std::string{f.value.get_string().value()};
+    // if (!f.value.is_string())
+    //   die("hard");
```

Comment:
Given that we are more lenient in the case that the `nesting-depth` can't be parsed to an int, I think it would be more consistent to also log an error here, and return the base type, assuming that we're seeing data from a newer VAST version that we don't understand, but at least we can work with the data itself.

Reply:
Correcting myself, since this happens inside a single key/value pair in the json, it's maybe even just a warning and we "lost" an attribute, but it's even less severe than returning the base type.